### DISCUSSION TITLE
文献总结与问答——🥷Paper Ninja

### DISCUSSION THREAD

**Discussion by hyacz** [InternLM/Tutorial#166] [project]
<p align="center">
  <a href="https://www.bing.com/images/create/origami-art2c-ninja-logo2c-4k/1-659b7f2669624ce1ba2f3489e54fab49?id=XkdiDhtG%2bXepg6X1tsiC4g%3d%3d&view=detailv2&idpp=genimg&FORM=GCRIDP&mode=overlay">
    <img width="250" height="250" src="https://github.com/InternLM/tutorial/assets/9551197/1702e843-81cf-4305-8b68-177be30ac54f">
  </a>
</p>

-------
# 🥷 Paper Ninja

## 项目背景
读论文和写论文是科研工作流中不可或缺的过程，正如吴恩达所言—— AI 即是新时代的电力，这场“新能源革命”也早已影响到科研工作流，催生了一批非常优秀的基于 LLM 的文献问答工具，例如 [ChatPaper](https://github.com/kaixindelele/ChatPaper) 、[gpt_academic](https://github.com/binary-husky/gpt_academic) 和 [PaperQA](https://github.com/whitead/paper-qa)。但由于速度，易用性和 APIKEY 等种种原因，笔者对于这些工具都是浅尝辄止，没能有效的将其整合到自己的工作流中来。作为一个练习项目，本项目不打算设计太高的目标，目前计划是结合上述工具与 InternLM-20B 模型，搭建一个可本地部署的文献总结与文献问答工具，欢迎感兴趣的朋友们一起参与。

## 功能设计
  - 基础：基于现有开源项目的 WebUI 或者 CLI 实现基本功能。
  - 进阶：从 Zotero API 获取个人文献库中的文献，以 Note 的形式输出到 Zotero 文献库。
### 文献总结
根据预设的 Prompt 对 PDF 格式文献进行总结。

### 文献问答
设置多篇文献作为上下文，构建向量数据库，对文献内容进行问答。

## 项目成员（18班）
- @hyacz
- @ggfengxuan
- @xwhclaire

**Comment by ggfengxuan** [comment]
求带飞~

**Comment by xwhclaire** [comment]
抱紧🦵

**Comment by JackieGLQ** [comment]
参与！

**Comment by szhowardhuang** [comment]
me参与

**Comment by siuserjy** [comment]
你好请问还需要队友吗

**Comment by Wanghlearn** [comment]
你好，请问还可以入对吗

**Comment by aJupyter** [comment]
你好，请问还能参加队伍吗

**Comment by hyacz** [comment]
@aJupyter @Wanghlearn @siuserjy @szhowardhuang @JackieGLQ 各位不好意思，大作业组队规则我之前没有理解对，刚和助教沟通了一下，不可以跨班级组队，我们这个项目是在 18 班的，大作业人数限制 5 人。

**Reply by siuserjy** [reply]
好的吧谢谢

**Comment by seifer08ms** [comment]
刚试了paperqa集成Internlm做论文的总结，就发现都有项目了，真快啊。
